File: Easy/55- 383. Ransom Note/383. Ransom Note.cs
public class Solution
{
    public bool CanConstruct(string ransomNote, string magazine)
    {
        if (ransomNote.Length > magazine.Length)
            return false;

        var map = new Dictionary<char, int>();

        foreach (char c in ransomNote)
        {
            if (map.ContainsKey(c))
                map[c]++;
            else
                map.Add(c, 1);
        }
        foreach (var c in magazine)
        {
            if (map.ContainsKey(c))
                map[c]--;
            else
                map.Add(c, -1);
        }

        foreach (var item in map)
        {
            if (item.Value > 0)
                return false;
        }
        return true;
    }
}

File: Easy/55- 383. Ransom Note/383. Ransom Note.js
/**
 * @param {string} ransomNote
 * @param {string} magazine
 * @return {boolean}
 */
var canConstruct = function (ransomNote, magazine) {
    if (ransomNote.length > magazine.length) return false;

    let map = new Map();
    for (let i = 0; i < ransomNote.length; i++) {
        if (map.has(ransomNote[i])) map[ransomNote[i]]++;
        else map.set(ransomNote[i], 1);
    }
    for (let i = 0; i < magazine.length; i++) {
        if (map.has(magazine[i])) map[magazine[i]]--;
        else map.set(magazine[i], -1);
    }
    for (const c of map.values()) {
        if (c > 0) return false;
    }
    return true;
};

let ransomNote = "aa",
    magazine = "aab";
console.log(canConstruct(ransomNote, magazine));
